src/App.tsx

```diff
@@ -61,14 +63,19 @@ export const App: React.FC = () => {
       filteredMarkets.map(currentMarket => {
         return getData(`mktDetail?id=${currentMarket.id}`)
           .then(response => checkForError(response))
-          .then(response => response.json());
+          .then(response => response.json())
+          .then(data => cleanDetailsData(data.marketdetails, currentMarket.id))
       })
     )
-      .then(arrayOfPromises => cleanDetailsData(arrayOfPromises))
-      .then(cleanData => setDetails(cleanData))
-      .catch(error => setError(error));
+    .then(data => addScheduleToMarkets(data, filteredMarkets))
```

Comment:
Could the schedule be added as part of the cleaning process? It's adding a lot of iterations to have the forEach inside of addScheduleToMarkets

Reply:
no. It can only happen _after_ both datasets are fetched. it's a double dataset problem. requires a nested iterator